sbi/samplers/rejection/rejection.py

```diff
@@ -131,7 +137,19 @@ def log_prob(self, theta: Tensor, **kwargs) -> Tensor:
 
         # To cover cases with few samples without leakage:
         sampling_batch_size = min(num_samples, max_sampling_batch_size)
+        start_time = time.time()
         while num_remaining > 0:
+            if (
+                max_sampling_time is not None
+                and (time.time() - start_time) > max_sampling_time
+            ):
+                raise RuntimeError(
```

Comment:
Nice. I would also mention that the user can set `reject_outside_prior_support` to `True` if they want to just get samples from the trained estimator.